WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderFollowCommentsHandler.kt

```diff
@@ -0,0 +1,71 @@
+package org.wordpress.android.ui.reader
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import org.wordpress.android.modules.BG_THREAD
+import org.wordpress.android.ui.pages.SnackbarMessageHolder
+import org.wordpress.android.ui.reader.usecases.ReaderCommentsFollowUseCase
+import org.wordpress.android.ui.reader.usecases.ReaderCommentsFollowUseCase.FollowCommentsState
+import org.wordpress.android.viewmodel.Event
+import javax.inject.Inject
+import javax.inject.Named
+import kotlinx.coroutines.flow.collect
+import org.wordpress.android.ui.reader.usecases.ReaderCommentsFollowUseCase.FollowCommentsState.Failure
+import org.wordpress.android.ui.reader.usecases.ReaderCommentsFollowUseCase.FollowCommentsState.FollowCommentsNotAllowed
+import org.wordpress.android.ui.reader.usecases.ReaderCommentsFollowUseCase.FollowCommentsState.FollowStateChanged
+import org.wordpress.android.ui.reader.usecases.ReaderCommentsFollowUseCase.FollowCommentsState.Loading
+import org.wordpress.android.ui.reader.usecases.ReaderCommentsFollowUseCase.FollowCommentsState.NoNetwork
+
+class ReaderFollowCommentsHandler @Inject constructor(
+    private val readerCommentsFollowUseCase: ReaderCommentsFollowUseCase,
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
+) {
+    private val _snackbarEvents = MediatorLiveData<Event<SnackbarMessageHolder>>()
+    val snackbarEvents: LiveData<Event<SnackbarMessageHolder>> = _snackbarEvents
+
+    private val _followStatusUpdate = MediatorLiveData<FollowCommentsState>()
+    val followStatusUpdate: LiveData<FollowCommentsState> = _followStatusUpdate
+
+    suspend fun handleFollowCommentsClicked(blogId: Long, postId: Long, askSubscribe: Boolean) {
+        withContext(bgDispatcher) {
```

Comment:
instead of this you should use `readerCommentsFollowUseCase.setMySubscriptionToPost(blogId, postId, askSubscribe).flowOn(bgDispatcher)`

Reply:
Thanks for sharing 🙇 . Done it in 612cc797d5ee09c2a1091ef9926fa0f47a35e476